pkg/plugins/backendplugin/backend_plugin.go

```diff
@@ -141,7 +141,14 @@ func (p *BackendPlugin) CollectMetrics(ctx context.Context) (*pluginv2.CollectMe
 		}, nil
 	}
 
-	res, err := p.diagnostics.CollectMetrics(ctx, &pluginv2.CollectMetricsRequest{})
+	var res *pluginv2.CollectMetricsResponse
+	err := InstrumentPluginRequest(p.id, "metrics", func() error {
+		var innerErr error
+		res, innerErr = p.diagnostics.CollectMetrics(ctx, &pluginv2.CollectMetricsRequest{})
+
+		return innerErr
+	})
+
```

Comment:
Nit, but do you mind removing this blank line? I think it reads much clearer if the error check is colocated with the originating call.

Reply:
Disagree. I like the code to breathe. 

For one-liners I agree ex.
```go
err := funcThatCanErr()
if err != nil {
    return err
}
```

But after `}` or `})` I always prefer some space.